### ISSUE TITLE
ZIPed SHP outputformat need WIDHT and HEIGHT

### ISSUE THREAD

**Issue by yjacolin** [mapserver/mapserver#5595]
Custom outputformat Mapfile contains this:
```
     OUTPUTFORMAT
       NAME "SHAPEZIP"
       DRIVER "OGR/ESRI Shapefile"
       FORMATOPTION "STORAGE=memory"
       FORMATOPTION "FORM=zip"
       FORMATOPTION "FILENAME=tadao_data.zip"
     END
```
When calling this outputformat in a WFS request, the second time, the request failed with this error:
```
msOGRWriteFromQuery(): General error message. Attempt to create directory '/vsimem/ogr_out/' failed.
msCalculateScale(): General error message. Invalid image width or height.
```

If I add `WIDTH=1&HEIGHT=1` to the request, it always works as expected.

**Comment by sdlime** [comment]
I assume if your set a SIZE in the mapfile the error also goes away? I'm wondering if this is really a bug or if it is a documentation issue.

**Comment by yjacolin** [comment]
Thanks for your tips, unfortunatly it doesn't seem to work with SIZE parameter in mapfile.

**Comment by jratike80** [comment]
Size and scales should have nothing to do with WFS. I wonder it the real issue is in the first message "General error message. Attempt to create directory '/vsimem/ogr_out/' failed." Does it exist already? Perhaps using WIDTH and HEIGHT, which are not WFS parameters and WFS server should just skip them quietly, has some side effect that makes it possible to create a new directory into vsimem.

It could also be worth checking why Mapserver goes the msCalculateScale() path with WFS, or is if perhaps a red herring?

Have you tested what happens if you change some WFS parameters, like maxFeatures (count in WFS 2)?
Does SHAPEZIP format work with disk storage?

**Comment by yjacolin** [comment]
> Size and scales should have nothing to do with WFS. I wonder it the real issue is in the first 
> message "General error message. Attempt to create directory '/vsimem/ogr_out/' failed." Does it 
> exist already? 

I agree, this looks strange. For me this is a virtual directory. @rouault could you confirm?

- [x] use of MAXFEATURE: NOK
- [ ] Does SHAPEZIP format work with disk storage?

I need to understand how to use / if SHAPEZIP format works with disk storage.

**Comment by jratike80** [comment]
http://mapserver.org/output/ogr_output.html
`FORMATOPTION "STORAGE=filesystem"`

**Comment by yjacolin** [comment]
- [ x] Does SHAPEZIP format work with disk storage? => OK

@jratike80 thanks :) So the issue come from memory storage.

**Comment by sdlime** [comment]
Actually WFS just wraps the msQueryBy... functions where scale computations can come into play - for layers with min/max scales set. This is especially true of the bounding box and point queries since those are often associated with with map interactions. What type of WFS request is being made?

**Comment by yjacolin** [comment]
@sdlime Here the working URL: http://xxx/mapserv?VERSION=1.1.0&SERVICE=WFS&TYPENAME=arret&REQUEST=GetFeature&OUTPUTFORMAT=SHAPEZIP&WIDTH=1&HEIGHT=1

**Comment by richie21** [comment]
I reproduce exactly the same problem, as described by @yjacolin, on MapServer 8.3-dev (from ms4w). With FORMATOPTION “STORAGE=filesystem”, it works. Thanks @jratike80.

Note that there was no problem with MapServer 6.4.1 (from ms4w).

**Comment by geographika** [comment]
@yjacolin @richie21 - test added for this in #7239, which successfully passes. If I remove `SIZE` from the Mapfile I can recreate the above error, however in the code this error would only appear if `SIZE` was missing, or removed somehow prior or during the request. 

https://github.com/MapServer/MapServer/blob/017074971d85c97cd8d4db80ee22de7f81b4259c/src/maptemplate.c#L278-L281

It looks like the WFS export code shares some of the template output code, but I'm not sure why the `checkScale` is required. It was added as part of the original implementation of https://mapserver.org/development/rfc/ms-rfc-62.html
